pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -1294,6 +1295,70 @@ func (c *Cluster) getActivatePVCInitContainer(osdProps osdProperties, osdID stri
 	return container
 }
 
+// var pvExpansion = `
+// osdSize=%d
+// pvcName=%s
+
+// pvcSize=$(kubectl get pvc $pvcName -o jsonpath='{.spec.resources.requests.storage}')
+// numericPvcSize="${pvcSize//[^0-9]/}"
+
+// # wait for the PVC to be expanded
+// while [ $numericPvcSize -ne $osdSize ]; do
+//   echo "PVC size not yet expanded, retrying in 2 seconds..."
+//   pvcSize=$(kubectl get pvc $pvcName -o jsonpath='{.spec.resources.requests.storage}')
+//   numericPvcSize="${pvcSize//[^0-9]/}"
+//   sleep 2
+// done
+
+// pvSize=$(kubectl get pv $(kubectl get pvc $pvcName -o jsonpath='{.spec.volumeName}') -o jsonpath='{.spec.capacity.storage}')
+// numericPvSize="${pvSize//[^0-9]/}"
+
+// # wait for the PV to be expanded
+// while [ $numericPvSize != $osdSize ]; do
+//   echo "PV size not yet expanded, retrying in 2 seconds..."
+//   pvSize=$(kubectl get pv $(kubectl get pvc $pvcName -o jsonpath='{.spec.volumeName}') -o jsonpath='{.spec.capacity.storage}')
+//   numericPvSize="${pvSize//[^0-9]/}"
+//   sleep 2
+// done
+// `
+
+var pvcExpansion = `
+osdSize=%d
+osdDataPath=%s
+
+device=$(lsblk -o NAME,SIZE,MOUNTPOINT | grep "$osdDataPath" | awk '{print $1}' | cut -c 3-)
+parent_disk=$(lsblk -no PKNAME "/dev/$device")
+disk_size=$(lsblk -b -o NAME,SIZE | grep "^$parent_disk" | awk '{print $2}')
+
+# add the check-pv-expansion init container to check if the PVC needs to be expanded
+# before the exand-bluefs init container runs
```

Comment:
why is this comment in the middle of the bash script? Isn't this more about how Rook sets it up than relevant to the script itself?